e2e/cephfs.go

```diff
@@ -848,6 +848,172 @@ var _ = Describe(cephfsType, func() {
 				}
 			})
 
+			By("Test subvolume snapshot and restored PVC metadata", func() {
+				err := createCephFSSnapshotClass(f)
+				if err != nil {
+					e2elog.Failf("failed to create CephFS snapshotclass: %v", err)
+				}
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC: %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+				err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC: %v", err)
+				}
+				snap := getSnapshot(snapshotPath)
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				// create snapshot
+				snap.Name = f.UniqueName
+				err = createSnapshot(&snap, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create snapshot (%s): %v", snap.Name, err)
+				}
+				_, pv, err := getPVCAndPV(f.ClientSet, pvc.Name, pvc.Namespace)
+				if err != nil {
+					e2elog.Failf("failed to get PV object for %s: %v", pvc.Name, err)
+				}
+				subVolumeName := pv.Spec.CSI.VolumeAttributes["subvolumeName"]
+				snaps, err := listCephFSSnapshots(f, fileSystemName, subVolumeName, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolume snapshots: %v", err)
+				}
+				if len(snaps) == 0 {
+					e2elog.Failf("cephFS snapshots list is empty %s/%s", fileSystemName, subVolumeName)
+				}
+				content, err := getVolumeSnapshotContent(snap.Namespace, snap.Name)
+				if err != nil {
+					e2elog.Failf("failed to get snapshotcontent for %s in namespace %s: %v",
+						snap.Name, snap.Namespace, err)
+				}
+				metadata, err := listCephFSSnapshotMetadata(f,
+					fileSystemName, subVolumeName, snaps[0].Name, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolume snapshots metadata: %v", err)
+				}
+				if metadata.VolSnapNameKey != snap.Name {
+					e2elog.Failf("failed, snapname expected:%s got:%s",
+						snap.Name, metadata.VolSnapNameKey)
+				} else if metadata.VolSnapNamespaceKey != snap.Namespace {
+					e2elog.Failf("failed, snapnamespace expected:%s got:%s",
+						snap.Namespace, metadata.VolSnapNamespaceKey)
+				} else if metadata.VolSnapContentNameKey != content.Name {
+					e2elog.Failf("failed, contentname expected:%s got:%s",
+						content.Name, metadata.VolSnapContentNameKey)
+				}
+				// Delete the parent pvc before restoring
+				// another one from snapshot.
+				err = deletePVCAndValidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to delete PVC: %v", err)
+				}
+
+				// Test Restore snapshot
```

Comment:
@Madhu-1 what kind of metadata details do we expect on the PVC which is restored from a snapshot?
I'm I missing anything below, I see the control is entering condition `if cloneMetadata.PVCNameKey != pvcClone.Name {}` all the time.

Reply:
check we have the metadata support in 17.2.1, which is the default in our CI

Reply:
Yes, we do! I'm rebasing this now, lets wait and see as you previously suggested.